### PR TITLE
UCP/CONTEXT: Add UCX_TLS aliases for cuda and rocm

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucp/core/ucp_context.c [hunk 1]
```diff
@@ -87,6 +87,8 @@ static ucs_config_field_t ucp_config_table[] = {
    " - ud      : ud_v and ud_x (preferably if available).\n"
    " - dc/dc_x : dc with accelerated verbs.\n"
    " - tcp     : sockets over TCP/IP.\n"
+   " - cuda    : Cuda (NVIDIA GPU) memory support.\n"
```
Comment (brminich) [inline src/ucp/core/ucp_context.c]: maybe `CUDA (NVIDIA GPU) memory support.\n`?

Comment (yosefe) [reply src/ucp/core/ucp_context.c]: > maybe `CUDA (NVIDIA GPU) memory support.\n`?

fixed

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 21 workers (click for details)</summary>

**Note:** the logs will be deleted after 01-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/56f556a0d5c89bf1f435125599b8305709f36412/_main) | :question: ABORTED
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/5b323234b4e5367dec5cff753e419e78a07a976a/hpc-arm-cavium-jenkins_W3) | :question: ABORTED
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/cf70a582c01925943fc581fe082b5aa4515b7bfc/hpc-arm-hwi-jenkins_W3) | :question: ABORTED
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/ba6f661fc3acd9b0fdb3b9790cd4969f3710c2a7/r-vmb-ppc-jenkins_W0) | :question: ABORTED
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/6ab8f61f54ab5ab860bab9462af4e28a41d3ba5b/r-vmb-ppc-jenkins_W1) | :question: ABORTED
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/625cca55db88fd52978bff2087b5bc74e17c4565/r-vmb-ppc-jenkins_W2) | :question: ABORTED
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/10c1432667b91650085b10ce4135a5189ba344da/r-vmb-ppc-jenkins_W3) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/9727b38be5456fa28c89f0fbee2c780a746aa4cc/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/580f18199b37c098ad2bb7bc2bdcf589ff16d139/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/25712f35bac02d6e9ff429ddfd7b93d19a8e9158/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/5e219c5ce058ecbc3cd3ea640f245af5500c3db6/hpc-arm-hwi-jenkins_W0) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/3024f9a648444942dc1a7af290faa9c9767adcd1/hpc-arm-hwi-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/9d245e1a947b3a67cff5949197cda86092b2b376/hpc-arm-hwi-jenkins_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/8c01618b3d621a66bd9f99867ccc21e2aa5fc608/hpc-test-node-gpu_W0) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/27a53087b97f0dc3661b6c9feed1ca63b0e3e627/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/469c22198835403eb13a4863945d573563abdccc/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/c75f5e9562d42c0dd4e4f45c0071d474679a9617/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/9941573f0399f9f2ede92405a6faa50056a8fa6e/hpc-test-node-new_W0) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/b5bbf41a9696e6ed67ac4e004efbb86f6e071321/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/331fa3c114730ed6e3a59fc1d1c4f898387a8ef2/hpc-test-node-new_W2) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/491de411d4a59788d3e6de8f40345421/raw/7ddefb883a1e19eaf56e4dcf822f9c96e164e902/hpc-test-node-new_W3) | :question: UNKNOWN

**Comment by yosefe** [comment]
> What if gdr_copy isn't configured on a system? Will `UCX_TLS=rc_x,cuda` fail on such systems, since gdr_copy TL wasn't found?

it would use other cuda tls

**Comment by dmitrygx** [comment]
> it would use other cuda tls

ok, thank you for clarification

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 01-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/39da3c96575586fb56de3674abe277c5f7921696/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/a061f02d43ea34689054b16178f0253816e159ed/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/f435858d0ca6c3b85f2bfe1d8bd023ca7f001fd0/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/d763daa7b9fb28ddb41ec128f1c9073fceafd324/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/e4579867d4f73cae0be48b253e749156cd634938/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/de3629d174faa1378bc679d4b1ac2086d50695b1/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/488cfba355d6f41451282a697760e94bacd9a8af/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/44cb54faa14141224b051c496408e0f4a1862bf9/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/91d7cdf63dc1d1d6defa963b59942df0016f6dff/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/2321e02767620f17505708fb3792ba59f9354a64/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/820252e359f7ccb4d4c96fa5487dc88e8b5a6159/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/a2960ff401bb7f14ae2bd1ae4012c0a649baff05/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/51590edfd65e78b0f26bd128ad4cd29770b21a19/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/7620b6e878057f92b82b78800bc6788eab44abbe/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/31089727e847af582cdd65e38a84034346114d9b/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/f097d6caa18b4689d270ffd8f1034fbcdba5d4bc/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/3f1d8148a5ec5c9a7b57dca739a4a4640fecb7c1/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/c501e663cdcfc5659dcf63362796539b3b189fba/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/a21c6db71ad2c9d193481229e1f120213117afdd/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/04c25447f6290888e894acfb3d4c83d59079a46c/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/2511482f10818b488f9930ea22b29d17641eddbd/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/59657f868f6a20da16098694f4cb9b56f124f898/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/a8c96b8c12c13a1b9c97f3c9c440789bd9fcd14f/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/aa15e5f80b6879ee3622e24b998f3bfc6f1ce148/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b44b4f3159e4b32bbdb1ed47447b48fc/raw/4e27c5d2f27c18558d6b060a8f2ee74f0e8c04de/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**